### PR TITLE
Refactored to regex

### PR DESCRIPTION
**Pull request by JackuB**
Removed `js-yaml`

### CODE REVIEW

**Comment by JackuB** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Tx5Bp0zX&#x2F;32-create-browser-build-for-decard-cain"> Create browser build for Decard Cain</a>

File: src/deckardcain.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import YAML from 'js-yaml';
-
 const API_BLUEPRINT_HEADER = /^[\uFEFF]?(((VERSION:( |\t)2)|(FORMAT:( |\t)(X-)?1A))([\n\r]{1,2}|$))/i;
-const API_BLUEPRINT_RESPONSE = /\+\s+Response\s+\d{3}/i;
+const API_BLUEPRINT_RESPONSE = /\+\s+(?:[Rr]esponse|[Rr]equest)\s+\d{3}/i;
```
Comment (honzajavorek) [inline src/deckardcain.js]: the `[Rr]` should be already handled by the `i` flag at the end

File: src/deckardcain.js [hunk 2]
```diff
@@ -13,51 +13,36 @@ const LEGACY_BLUEPRINT_TITLE = /\-{3} ([^\n\r]+ )?\-{3}([\n\r]{1,2}|$)/;
...
+  }
+
+  if (source.match(API_BLUEPRINT_RESPONSE)) {
+    // I can not see the '--- Sample Title ---' and at the same time
+    // there is something like '+ Response 200' in the document, which is
+    // pretty distinctive for API Blueprint.
+    return 'text/vnd.apiblueprint';
+  }
+
+  return null;
```
Comment (honzajavorek) [inline src/deckardcain.js]: Well the flow of this file now looks almost like one big switch/case. Although I crafted the comments to look like Deckard's speach, I think it's not worth to keep them in that form anymore and we should simplify it so it's more readable:

```js
if (source.match(API_BLUEPRINT_RESPONSE)) {
  // not '--- Sample Title ---' and '+ Response 200'
  return 'text/vnd.apiblueprint';
}
```

Comment (JackuB) [inline src/deckardcain.js]: I'll do patch PR for regex and this if you don't mind me removing your comments.

Comment (honzajavorek) [inline src/deckardcain.js]: Already addressed by @freaz I guess, but feel free to alter the comments or move them around. I don't have feelings attached to them and they don't help collaboration or maintainability in their "Deckard Cain speaks" form :)

**Pull request merged**